### PR TITLE
SALTO-1198: saving the last fetch time per instance

### PR DESCRIPTION
**Pull request by alonstern**
Changed to save the last fetch time per instance.

This is done so when making two partial fetches in a row, if the element that was changed in the service was not fetched in the first fetch, it would be fetched in the second and vice versa.

I do this by saving on each instance the last time it was fetched (with the server time) and saving the last time of the full fetch.
Then I make all the queries using the full fetch time and filter from the results elements that were already fetched using the time I saved on each element.

The whole `element_source_index` part in this PR is to avoid iterating over the elementSource more than once (once the lazy loading feature will be merged, it can take some time in big workspaces).

---
_Release Notes_: 
None

### CODE REVIEW

File: packages/netsuite-adapter/src/adapter.ts [hunk 1]
```diff
@@ -126,7 +130,11 @@ export default class NetsuiteAdapter implements AdapterOperations {
     ].filter(values.isDefined).reduce(andQuery)
 
 
-    const { serverTimeElements, changedObjectsQuery } = await this.runSuiteAppOperations(fetchQuery)
+    const {
+      serverTimeElements,
```
Comment (omrilit) [inline packages/netsuite-adapter/src/adapter.ts:134]: nit: you can call `createServerTimeElements` from here using the `serverTime`, and remove the related logic in `runSuiteAppOperations`, your call

File: packages/netsuite-adapter/src/adapter.ts [hunk 2]
```diff
@@ -171,31 +181,42 @@ export default class NetsuiteAdapter implements AdapterOperations {
...
 
     if (this.fetchTarget === undefined) {
-      return { serverTimeElements: createServerTimeElements(sysInfo.time) }
+      return {
+        serverTimeElements: createServerTimeElements(sysInfo.time),
+        serverTime: sysInfo.time,
+      }
     }
 
     const lastFetchTime = await getLastServerTime(this.elementsSource)
```
Comment (omrilit) [inline packages/netsuite-adapter/src/adapter.ts:206]: Should this one use `elementsSourceIndex`?

File: packages/netsuite-adapter/src/changes_detector/changes_detector.ts [hunk 3]
```diff
@@ -128,7 +167,8 @@ export const getChangedObjects = async (
     isObjectMatch: objectID => !SUPPORTED_TYPES.has(objectID.type)
       || scriptIds.has(objectID.scriptId)
       || types.has(objectID.type),
-    isFileMatch: filePath => paths.has(filePath),
+    isFileMatch: filePath =>
+      paths.has(filePath),
```
Comment (omrilit) [inline packages/netsuite-adapter/src/changes_detector/changes_detector.ts:171]: I guess this superrrr nit wasn't in purpose

**Review by omrilit — APPROVED** [review]
Had only few minors. looks great!

File: packages/netsuite-adapter/src/adapter.ts (hunk 2 above)
Comment (alonstern) [reply packages/netsuite-adapter/src/adapter.ts:206]: Not really, since it gets a specific element and does not iterate over all the elements

**Pull request merged**